internal/cli/csrconfig.go

```diff
@@ -0,0 +1,126 @@
+package cli
+
+import (
+	"bufio"
+	"context"
+	"encoding/base64"
+	"fmt"
+	"os"
+
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+)
+
+type CSRConfigOptions struct {
+	GlobalOptions
+	CertSigningRequestFile string
+	Outfile                string
+	Name                   string
+	ExpirationSeconds      string
+	Overwrite              bool
+}
+
+func DefaultCSRConfigOptions() *CSRConfigOptions {
+	return &CSRConfigOptions{
+		GlobalOptions: DefaultGlobalOptions(),
+		Overwrite:     false,
+	}
+}
+
+func NewCmdCSRConfig() *cobra.Command {
+	o := DefaultCSRConfigOptions()
+	cmd := &cobra.Command{
+		Use:   "csr-generate",
+		Short: "generate a CSR resource config .yaml based on a CSR file .csr",
+		Args:  cobra.ExactArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if err := o.Complete(cmd, args); err != nil {
+				return err
+			}
+			if err := o.Validate(args); err != nil {
+				return err
+			}
+			return o.Run(cmd.Context(), args)
+		},
+		SilenceUsage: true,
+	}
+	o.Bind(cmd.Flags())
+	_ = cmd.MarkFlagRequired("output")
+	return cmd
+}
+
+func (o *CSRConfigOptions) Bind(fs *pflag.FlagSet) {
+	o.GlobalOptions.Bind(fs)
+	fs.StringVarP(&o.Outfile, "output", "o", "", "specify a filename in which to save the generated csr resource config")
+	fs.StringVarP(&o.Name, "name", "n", "mycsr", "specify a name for the csr")
+	fs.StringVarP(&o.ExpirationSeconds, "expiry", "e", "604800", "specify desired certificate expiration in seconds")
+	fs.BoolVarP(&o.Overwrite, "overwrite", "y", false, "setting this flag overwrites the specified output file without prompting the user")
+}
+
+func (o *CSRConfigOptions) Complete(cmd *cobra.Command, args []string) error {
+	return o.GlobalOptions.Complete(cmd, args)
+}
+
+func (o *CSRConfigOptions) Validate(args []string) error {
+	if err := o.GlobalOptions.Validate(args); err != nil {
+		return err
+	}
+
+	if len(args) == 0 {
+		return fmt.Errorf("must specify a CSR file")
+	}
+	_, err := os.ReadFile(args[0])
+	if err != nil {
+		return fmt.Errorf("invalid CSR file specified: %w", err)
+	}
+
+	return nil
+}
+
+func (o *CSRConfigOptions) Run(ctx context.Context, args []string) error {
+	if len(args) == 0 {
+		return fmt.Errorf("must specify a CSR file")
+	}
+
+	data, err := os.ReadFile(args[0])
+	if err != nil {
+		return fmt.Errorf("invalid CSR file specified: %w", err)
+	}
+
+	encoded := base64.StdEncoding.EncodeToString(data)
+
+	output := fmt.Sprintf(`apiVersion: v1alpha1
+kind: CertificateSigningRequest
+metadata:
+  name: %s
+spec:
+  request: %s
+  signerName: ca
+  usages: ["clientAuth", "CA:false"]
+  expirationSeconds: %s
+`, o.Name, encoded, o.ExpirationSeconds)
+
+	if !o.Overwrite {
```

Comment:
I'd rather not make the CLI interactive.  I'm OK with the overwrite flag for extra safety, but if the user didn't pass it, just return an error.

Reply:
Done